// @pivotal-tools/commons-vscode/lib/highlight-service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const VSCode = require("vscode");
const path = require("path");
function toDecoration(rng) {
    return new VSCode.Range(toPosition(rng.start), toPosition(rng.end));
}
function toPosition(p) {
    return new VSCode.Position(p.line, p.character);
}
class HighlightService {
    dispose() {
        this.DECORATION.dispose();
    }
    constructor() {
        this.DECORATION = VSCode.window.createTextEditorDecorationType({
            //        textDecoration: "underline",
            gutterIconPath: path.resolve(__dirname, "../icons/boot-icon.png"),
            gutterIconSize: "contain",
            outline: "#32BA56 dotted thin"
        });
        this.highlights = new Map();
    }
    handle(params) {
        this.highlights.set(params.doc.uri, params.ranges);
        this.refresh(params.doc.uri);
    }
    refresh(uri) {
        let editors = VSCode.window.visibleTextEditors;
        for (let editor of editors) {
            let activeUri = editor.document.uri.toString();
            if (uri === activeUri) {
                //We only update highlights in the active editor for now
                let highlights = this.highlights.get(uri) || [];
                let decorations = highlights.map(hl => toDecoration(hl));
                editor.setDecorations(this.DECORATION, decorations);
            }
        }
    }
}
exports.HighlightService = HighlightService;
